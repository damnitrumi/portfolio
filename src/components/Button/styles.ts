import styled, { css } from "styled-components";

export const Wrapper = styled.button`
    ${() => css`
      border: none;
      background-color: transparent;
      cursor: pointer;
    `}
`;
