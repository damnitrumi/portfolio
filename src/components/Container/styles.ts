import styled, { css } from "styled-components";

export const Wrapper = styled.div`
    ${({ theme }) => css`
        background-color: ${theme.colors.primaryColor};
        padding: ${theme.paddings.containerDefault};
        height: 100vh;

        @media ${theme.media.ltaMedium}{
            padding: ${theme.paddings.containerLtaMedium};
        }
    `}
`;
