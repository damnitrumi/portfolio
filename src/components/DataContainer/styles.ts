import styled, { css } from "styled-components";

export const Wrapper = styled.div`
    ${() => css`
        max-width: 1100px;
        margin: 0 auto;
        height: 100%;
        padding: 200px 0;
    `}
`;
