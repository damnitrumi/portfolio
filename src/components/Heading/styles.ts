import styled, { css } from "styled-components";
import { HeadingProps } from ".";

const pickSize = (size: string) => {
  if (size) {
    return css`
      font-size: ${size};
    `;
  }
};

export const Title = styled.h1<HeadingProps>`
${({ size, as }) => css`
  color: white;
  ${pickSize(size)}
  font-weight: ${as == "h1" ? 500 : 400};
`}
`;
