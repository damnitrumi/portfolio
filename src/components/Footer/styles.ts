import styled, { css } from "styled-components";
import { Wrapper as Container } from "../Container/styles";

export const Wrapper = styled.div`
  ${() => css`
  `}
`;

export const FooterContainer = styled(Container)`
    background-color: transparent;
    height: auto;
    position: fixed;
    width: 100%;
    bottom: 0;
`;

export const LinksContainer = styled.div``;
