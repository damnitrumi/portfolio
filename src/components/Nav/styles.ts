import styled, { css } from "styled-components";
import { Wrapper as Container } from "../Container/styles";

export const Wrapper = styled.nav`
    ${() => css`
        display: flex;
        align-items: center;
        justify-content: space-between;
    `}
`;

export const imgContainer = styled.a`
    ${() => css`
        max-width: 150px;
    `}
`;

export const NavContainer = styled(Container)`
    background-color: transparent;
    height: auto;
    position: fixed;
    width: 100%;
    z-index: 1;

    @media ${({ theme }) => theme.media.ltaMedium}{
      background-color: ${({ theme }) => theme.colors.mediaGray};
    }
`;

export const bgMenu = styled.div`
    position: fixed;
    height: 100vh;
    width: 100%;
    top: 0;
    left: 0;
    background-color: rgba(0, 0, 0, 0.3);
    /* pointer-events: none; */
`;
