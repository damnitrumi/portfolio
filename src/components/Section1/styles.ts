import styled, { css } from "styled-components";
import { Title as Heading } from "../Heading/styles";
import { Wrapper as Text } from "../Text/styles";

export const Wrapper = styled.div`
    ${({ theme }) => css`
      @media ${theme.media.ltaMedium}{
        ${Heading}{
          font-size: 8.0rem;
        }
        ${Text}{
          font-size: 1.6rem;
        }
      }
    `}
`;
