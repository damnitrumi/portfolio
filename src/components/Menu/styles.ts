import styled, { css } from "styled-components";
import { MenuBehaviorProps } from ".";
import { LinksContainer as IconsContainerMock } from "../Footer/styles";
import { StyledIconBase } from "@styled-icons/styled-icon";

export const Wrapper = styled.div<MenuBehaviorProps>`
    ${({ theme, menuVisible }) => css`
    position: fixed;
    width: 100%;
    max-width: 500px;
    top: 0;
    right: -100%;
    transition: all .8s ease-in-out;

    @media ${theme.media.ltaMedium} {
      max-width: 100%;
    };

    ${
      menuVisible &&
      css`
        right: 0;
    `
    }
    `}
`;

export const ButtonContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`;
export const LinksContainer = styled.div`
  padding: 30px 0;

  a{
    padding: 20px 0;
    color: ${({ theme }) => theme.colors.menuGray};
    transition: all 200ms ease-in;
  }

  a:hover{
    color: ${({ theme }) => theme.colors.white};
  }
`;

export const IconsContainer = styled(IconsContainerMock)`
  ${StyledIconBase}{
    color: ${({ theme }) => theme.colors.menuGray};
    &:hover{
      color: ${({ theme }) => theme.colors.hoverGreen};
    }
  }
`;
