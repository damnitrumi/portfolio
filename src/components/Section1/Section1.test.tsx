import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { Section1 } from ".";

describe("<Section1 />", () => {
  it("should render Section1", () => {
    renderTheme(<Section1 />);
    expect(screen.getByRole("heading")).toHaveStyleRule("font-size", "9.2rem");
    expect(screen.getByText("DESENVOLVEDOR FRONT-END")).toHaveStyleRule(
      "font-size",
      "2.2rem",
    );
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(<Section1 />);
    expect(container).toMatchSnapshot();
  });
});
