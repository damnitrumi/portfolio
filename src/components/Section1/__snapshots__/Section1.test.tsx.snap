// Vitest Snapshot v1

exports[`<Section1 /> > should match snapshot 1`] = `
.c1 {
  background-color: #1b1b1b;
  padding: 30px 60px;
  height: 100vh;
}

.c2 {
  max-width: 1100px;
  margin: 0 auto;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  color: #FFFFFF;
  font-size: 9.2rem;
  font-weight: 500;
}

.c6 {
  color: #FFFFFF;
  font-size: 2.2rem;
  line-height: 30px;
}

@media (max-width:768px) {
  .c1 {
    padding: 20px;
  }
}

@media (max-width:768px) {
  .c0 .c3 {
    font-size: 8.0rem;
  }

  .c0 .c5 {
    font-size: 1.6rem;
  }
}

<div>
  <div
    class="c0"
    id="home"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3 c4"
        />
        <p
          class="c5 c6"
        >
          DESENVOLVEDOR FRONT-END
        </p>
      </div>
    </div>
  </div>
</div>
`;
